docs/content/en/docs/tutorials/ci_cd.md

```diff
@@ -0,0 +1,286 @@
+
+
+## Step 7: Getting the CI pipeline setup
+
+Now that we have skaffold ready and working locally, the CI pipeline requires little to no effort since it will just call skaffold to do all the builds and deployments.
+
+Here is a sample pipeline using Gitlab CI
+
+
+```
+stages:
+    - my-stage
+
+my-job:
+    image:
+        name: gcr.io/k8s-skaffold/skaffold:v1.15.0
+    tags:
+        - development
+    stage: my-stage
+    retry: 2
+    script:
+        - docker login $REG_DEV_URL -u $REG_DEV_UNAME -p $REG_DEV_PASSWORD
+        - echo "$GCP_DEV_SERVICE_KEY" > gcloud-service-key.json
+        - gcloud auth activate-service-account --key-file gcloud-service-key.json
+        - gcloud config set project $PROJECT_DEV_NAME
+        - gcloud config set compute/zone $PROJECT_DEV_REGION
+        - gcloud container clusters get-credentials $CLUSTER_DEV_NAME
+        - kubectl config get-contexts
+        - DOCKER_HOST="$DIND_DEV" NPM_REGISTRY="$NPM_REGISTRY_DEV" NPM_TOKEN="$NPM_TOKEN_DEV" skaffold run --kube-context $CLUSTER_DEV_CONTEXT -n default -l skaffold.dev/run-id=deploydep -p dev-svc --status-check
+    only:
+        refs:
+            - master
+```
+
+
+Before we look at the pipeline, you might want to look at Gitlab CI/CD environment variables here: [https://docs.gitlab.com/ee/ci/variables/](https://docs.gitlab.com/ee/ci/variables/)
+
+This is how we will configure the various environment variables to be passed in to the pipeline. Make sure that you mask/protect them using Gitlab so that they are not visible in the logs.
+
+Here, we are:
+
+
+
+*   Logging into the docker registry specifying the Registry URL, User ID and password (You can choose to authenticate using service tokens as well or other auth providers as well)
+*   Logging in to our gcp account using the service account key
+*   Specifying the project, zone, cluster to deploy our application in.
+*   We are using DIND (Docker In Docker) to facilitate our builds (more about that below)
+*   Then we just call the skaffold run in the pipeline passing flags we want to pass at runtime (in our case our cluster context, env vars, namespace and some labels)
+*   And we trigger this only on changes to master branch
+
+So, ultimately our Gitlab CI pipeline does not know anything about the helm charts or build strategy or any deployment methods and relies completely on skaffold to do the job for us.
+
+This works equally well irrespective of whether you choose the Docker Executor ([https://docs.gitlab.com/runner/executors/docker.html](https://docs.gitlab.com/runner/executors/docker.html)) or the Kubernetes Executor ([https://docs.gitlab.com/runner/executors/kubernetes.html](https://docs.gitlab.com/runner/executors/kubernetes.html)) and would require little to no changes.
```

Comment:
This assumes some knowledge about Gitlab.  I *think* you're using the Kubernetes executor here (since you have a deployment yaml below, which doesn't look related to your application).  Why did you choose this if you could use the Docker Executor?

Reply:
Docker executor does not run in the Kubernetes cluster, requires access to the Docker engine and also does not allow for things like scaling up/down runners as pods in the cluster all of which is supported in Kubernetes executor. While all the instructions mentioned in this tutorial will work in both runners, Kubernetes executor has its own set of advantages. Should I elaborate about it here?

Reply:
I think the example should use the simplest approach (which I'm guessing is the Docker executor).  You could include a pointer to the Kubernetes executor and briefly summarize some of its advantages.